python/tvm/contrib/hexagon/session.py

```diff
@@ -95,6 +94,25 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, exc_traceback):
         pass
 
+    @property
+    def device(self):
+        """Session device."""
+
+        if hasattr(self, "_device") and self._device is not None:
+            return self._device
+
+        if not hasattr(self, "_requires_cpu_device"):
```

Comment:
Do we want to supply a default for `_requires_cpu_device`, rather than requiring it to be explicitly set?  I think general case is that we return a "hexagon" device, and that AOT is the special case that would require a CPU device.  Rather than requiring it to be set for both paths, can we set `self._requires_cpu_device = False` in the initializer, and only set it to True in the AOT path.  That way, the default is to return a "hexagon" device.

Also, as a general nitpick, I'd recommend `assert(condition)` instead of `if not condition: assert(False)`.

Reply:
Current CI results have some failures for `test_2d_physical_buffers` and `test_cache_read_write`, which look to be caused by this lack of a default.  https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-11065/8/tests